semantic_release/history/logs.py

```diff
@@ -32,17 +32,31 @@ def evaluate_version_bump(current_version, force=None):
     changes = []
     commit_count = 0
 
-    for commit_message in get_commit_log('v{0}'.format(current_version)):
-        if current_version in commit_message:
-            break
-
-        try:
-            message = current_commit_parser()(commit_message)
-            changes.append(message[0])
-        except UnknownCommitMessageStyleError:
-            pass
-
-        commit_count += 1
+    if (config.has_option('semantic_release', 'versioning_by_tag') and
+            config.getboolean('semantic_release', 'versioning_by_tag')):
+        # we have to first find our the version hash tagged correspondingly
+        version_hash = get_version_from_tag('v{0}'.format(current_version))
+        for commit_message in get_commit_log(version_hash):
```

Comment:
This is the same as `for commit_message in get_commit_log('v{0}'.format(current_version)):`, thus I think it is unnecessary to get the hash first. If you change this back, the only difference between tag and regular here is that the version will not end up in the commit message so the break won't be called for tag, but that is equal to the other case here. I think this can be changed back to the way it was and still work with the tag version.